File: version_java/src/main/java/com/example/version_java/ui/home/HomeViewModel.java
package com.example.version_java.ui.home;

import android.app.Application;

import com.example.version_java.base.BaseViewModel;

import org.jetbrains.annotations.NotNull;

import javax.inject.Inject;

import dagger.hilt.android.lifecycle.HiltViewModel;

@HiltViewModel
public class HomeViewModel extends BaseViewModel {

    public void permissionGrant() {
        viewStateChanged(HomeViewState.PermissionGrant.INSTANCE);
    }

    @Inject
    HomeViewModel(@NotNull Application application) {
        super(application);
    }

    ;
}

File: app/src/main/java/com/example/crimenotification/ui/home/HomeViewModel.kt
package com.example.crimenotification.ui.home

import android.app.Application
import com.example.crimenotification.base.BaseViewModel
import dagger.hilt.android.lifecycle.HiltViewModel
import javax.inject.Inject

@HiltViewModel
class HomeViewModel @Inject constructor(app: Application) : BaseViewModel(app){

    /**
     * 퍼미션 허용했을때 상태값이 PermissionGrant 변한다.
     */
    fun permissionGrant() {
        viewStateChanged(HomeViewState.PermissionGrant)
    }

}